File: python/leap/leap.py
def divisible(num, div):
    """Return True if num is divisible by div"""
    return num % div == 0


def is_leap_year(year):
    """Returns True if year is a leap year

    A year is a leap year if it is divisible by 4, unless it is divisible by
    100, unless it is also divisible by 400.
    """
    return divisible(year, 4) and not divisible(year, 100) or divisible(year,
                                                                        400)

File: go/leap/leap.go
// Package leap provides utilities for leap years
package leap

const testVersion = 3

// IsLeapYear returns true if year is a leap year
func IsLeapYear(year int) bool {
	multiple := func(x int) bool {
		return divisibleBy(year, x)
	}
	return multiple(4) && (!multiple(100) || multiple(400))
}

func divisibleBy(x, y int) bool {
	return x%y == 0
}
